### ISSUE TITLE
Modal views aren't styled when using QtView and style_sheet.

### ISSUE THREAD

**Issue by jmilloy** [enthought/traitsui#286]
First, I would prefer to use the Enthought-dev mailing list, but the link does not seem to be working.

I would like to be able to style a modal `QtView`. The following should demonstrate the issue:

```
from traits.api import Str
from traitsui.api import Handler, Label
from traitsui.menu import OKButton
from traitsui.qt4.extra.qt_view import QtView

class Alert(Handler):
    """ Non-modal window with working styling. """

    msg = Str('Sample message')

    def traits_view(self):
        style_sheet = '''
            QLabel {
                font: 36pt "Verdana";
                margin-left: 12px;
            }
            QPushButton {
                font: 36pt "Verdana";
                margin: 16px;
                padding: 8px
            }
            '''

        return QtView(
            Label(self.msg),
            buttons=[OKButton],
            height=1.0,
            width=1.0,
            style_sheet=style_sheet,
            )

class AlertModal(Handler):
    """ Modal window with styling error. """

    msg = Str('Sample message')

    def traits_view(self):
        style_sheet = '''
            QLabel {
                font: 36pt "Verdana";
                margin-left: 12px;
            }
            QPushButton {
                font: 36pt "Verdana";
                margin: 16px;
                padding: 8px
            }
            '''

        return QtView(
            Label(self.msg),
            buttons=[OKButton],
            kind='modal',
            height=1.0,
            width=1.0,
            style_sheet=style_sheet,
            )

# styling works
alert = Alert()
alert.configure_traits()

# no styling, produces an exception when the window is closed
alert = AlertModal()
alert.configure_traits()
```

The modal window raises the following Exception after closing, in which the `ui.control` attribute is None so the style sheet can't be set:

```
Traceback (most recent call last):
  File "test3.py", line 61, in <module>
    alert.configure_traits()
  File "/usr/local/lib/python2.7/dist-packages/traitsui/handler.py", line 459, in configure_traits
    scrollable, **args )
  File "/usr/local/lib/python2.7/dist-packages/traits/has_traits.py", line 2156, in configure_traits
    kind, handler, id, scrollable, args )
  File "/usr/local/lib/python2.7/dist-packages/traitsui/qt4/toolkit.py", line 213, in view_application
    id, scrollable, args )
  File "/usr/local/lib/python2.7/dist-packages/traitsui/qt4/view_application.py", line 83, in view_application
    scrollable, args ).ui.result
  File "/usr/local/lib/python2.7/dist-packages/traitsui/qt4/view_application.py", line 134, in __init__
    args       = self.args )
  File "/usr/local/lib/python2.7/dist-packages/traitsui/qt4/extra/qt_view.py", line 57, in ui
    ui.control.setStyleSheet(self.style_sheet)
AttributeError: 'NoneType' object has no attribute 'setStyleSheet'
```

**Comment by jmilloy** [comment]
traits 4.5.0
traitsui 4.5.1 and 5.0.0

**Comment by corranwebster** [comment]
It looks like a bug, so this is an appropriate place to report it.

As a work-around which may or may not work for your use case, have you tried using the Pyface `MessageDialog` instead?  It provides a standard Qt modal dialog, and you could then presumably apply whatever styling you need to the dialog control (an you may not need it, depending on your use-case)?  Eg.
```
from pyface.api import MessageDialog

dialog = MessageDialog(
    message='Sample message',
    informative="This is a sample message",
    detail="This is even more information which is hidden by default",
    severity='warning'
)
# normally you'd just do
# result = dialog.open()
# but for stylesheets you might need to hack it a bit (or subclass appropriately)
# dialog._create()
# dialog.control.setStyleSheet(stylesheet)
result = dialog.open()
```

**Comment by yves-surrel** [comment]
I propose a fix in #314

**Comment by corranwebster** [comment]
Closing in favour of #314.
